homeassistant/components/modbus/switch.py

```diff
@@ -111,32 +116,30 @@ async def async_turn_on(self, **kwargs):
         result = await self._hub.async_pymodbus_call(
             self._slave, self._address, self._command_on, self._write_type
         )
-        if result is False:
+        if result is None:
             self._available = False
```

Comment:
Cleanup slightly.
```suggestion
        self._available = result is not None
```

Reply:
good catch, I solved it a bit different, to avoid multiple if.